wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -217,159 +146,181 @@ class PresentationActivity : FragmentActivity() {
+                    /**
+                     * Device request bytes have been transmitted via device retrieval helper listener,
+                     * Start the Presentment Flow where a Presentment is shown for every
+                     * [DocumentRequest] that has a suitable [Document].
+                     */
+                    lifecycleScope.launch {
+
+                        try {
+                            val deviceRequest = DeviceRequestParser(
+                                deviceRequestByteArray!!,
+                                deviceRetrievalHelper!!.sessionTranscript
+                            ).parse()
+
+                            deviceRequest.docRequests.forEach { docRequest ->
+                                // find a suitable Document for the docRequest
+                                val document = findSuitableDocumentForRequest(docRequest)
+                                // if document == null no suitable documents could be found then skip to next docRequest
+                                val mdocCredential = document.findCredential(
+                                    WalletApplication.CREDENTIAL_DOMAIN_MDOC,
+                                    Clock.System.now()
+                                ) as MdocCredential
+
+                                // extract the TrustPoint if possible
+                                var trustPoint: TrustPoint? = null
+                                if (docRequest.readerAuthenticated) {
+                                    val result = walletApp.trustManager.verify(
+                                        docRequest.readerCertificateChain!!.javaX509Certificates,
+                                        customValidators = emptyList()  // not needed for reader auth
                                     )
+                                    if (result.isTrusted && result.trustPoints.isNotEmpty()) {
+                                        trustPoint = result.trustPoints.first()
+                                    } else if (result.error != null) {
+                                        Logger.w(
+                                            com.android.identity_credential.wallet.presentation.TAG,
+                                            "Error finding TrustPoint for reader auth",
+                                            result.error!!
+                                        )
+                                    }
                                 }
-                        }
-
-                        State.RESPONSE_SENT -> {
-                            stateDisplay.value = "Response Sent"
-                            Logger.i(TAG, "State: Response Sent")
-                        }
 
-                        else -> {}
-                    }
-                }
+                                // generate the DocumentRequest from the current docRequest
+                                val documentRequest = MdocUtil.generateDocumentRequest(docRequest)
```

Comment:
Nit: Moving `documentRequest = MdocUtil.generateDocumentRequest(docRequest)` into the showPresentmentCall() would save a couple of lines!